src/urllib3/contrib/socks.py

```diff
@@ -66,23 +67,37 @@
 try:
     import ssl
 except ImportError:
-    ssl = None
+    ssl = None  # type: ignore
+
+try:
+    from typing import TypedDict
+except ImportError:
+    from typing_extensions import TypedDict  # Python 3.7
+
+
+class SocksOptions(TypedDict):
```

Comment:
We may have to define this structure under the `try: from typing import TypedDict` section. Since it's not a public type we can name it `_TYPE_SOCKS_OPTIONS`.